snuba/web/db_query.py

```diff
@@ -350,11 +347,19 @@ def execute_query_with_readthrough_caching(
     query_id = get_query_cache_key(formatted_query)
     query_settings["query_id"] = query_id
 
+    span = Hub.current.scope.span
```

Comment:
Are you sure neither `current` nor `scope` can ever be null ? 
Generally we get the current scope as a context:

`with sentry_sdk.configure_scope() as scope:`

Reply:
It can't be null, since this function is wrapped by the `with_span` function, which would ensure there was an active span.

However the more I read the docs, the more I wonder whether I should actually make this a tag value. Then we could filter/search by transactions in Discover. That would probably be a useful thing to do as we start to evaluate query performance. Thoughts?

Reply:
> Then we could filter/search by transactions in Discover

THey would not. You would have to asign the tag to the main span and not to the current one. Like we do in views.py where we ensure we only set tags in the first span of the transaction (and it is very fragile)
https://github.com/getsentry/snuba/blob/master/snuba/web/views.py#L325-L329

Reply:
What happens if I do `sentry_sdk.set_tag()`? Doesn't that work for transactions as well?

Reply:
That one I don't know.

Reply:
I tested this: it does set the tag on the transaction, and we can then search for transactions with a specific status. Sentry really is an easy and intuitive product to use.

Reply:
What if the splitter splits the query and one is cached while the other is not ?

Reply:
Then the tag will be at worst half right. It's not ideal, but is better than we have now.